pss/trojan_chunk.go

```diff
@@ -0,0 +1,214 @@
+// Copyright 2020 The Swarm Authors
+// This file is part of the Swarm library.
+//
+// The Swarm library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The Swarm library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the Swarm library. If not, see <http://www.gnu.org/licenses/>.
+
+package pss
+
+import (
+	"crypto/rand"
+	"encoding/binary"
+	"encoding/json"
+	"math/big"
+	"reflect"
+
+	"github.com/ethersphere/swarm/chunk"
+	"github.com/ethersphere/swarm/storage"
+)
+
+// TODO: can we re-use some existing types here?
+type trojanHeaders struct {
+	span  []byte
+	nonce []byte
+}
+
+// TODO: can we re-use some existing types here?
+type trojanMessage struct {
+	length  []byte
+	topic   []byte
+	payload []byte
+	padding []byte
+}
+
+type trojanData struct {
```

Comment:
these are also not needed

Reply:
isn't it easier to serialize all the trojan data if we have a `struct` that encapsulates? then call the hash on this and compare it to the address. so i would at least go for 

```golang
type trojanChunk struct{
    address chunk.Address
    trojanMessage
}

type trojanMessage struct{
    // all message fields
}
````

Reply:
see [previous reply](https://github.com/ethersphere/swarm/pull/2148#discussion_r404461093)